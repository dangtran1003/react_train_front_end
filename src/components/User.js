import React, { Component} from 'react';
import PropTypes from 'prop-types'
import {user_del} from '../actions/user'
import { connect } from 'react-redux';

class User extends Component {
  static propTypes = {
    username : PropTypes.string,
    name : PropTypes.string,
    email : PropTypes.string,
    id: PropTypes.number
  }

  render() {
    const {username, name, email, id} = this.props
    return (
        <tr>
        <td>{username}</td>
        <td>{name}</td>
        <td>{email}</td>
        <td style={{width: 100}}>
          <a href={"/modify/"+id.toString()} className="btn btn-info" role="button" style={{margin:4}}><i className="icons font-l d-block mt-0 cui-wrench"></i></a>
          <button className="btn btn-danger" value={id} onClick={this.props.user_del}><i className="icons font-l d-block mt-0 cui-trash"></i></button>
        </td>
      </tr>
    )
  }
}

// export default User

const mapDispatchToProps = {
  user_del: user_del,
}

export default connect(null, mapDispatchToProps)(User)