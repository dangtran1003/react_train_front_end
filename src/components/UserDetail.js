import React, { Component} from 'react';
import PropTypes from 'prop-types'
import { connect } from 'react-redux';
import { user_detail, change_detail } from '../actions/user';

export class UserDetail extends Component {
  static propTypes = {
    email : PropTypes.string,
    name : PropTypes.string,
  }
  state = {
      email: '',
      name : '',
      error : ''
  }

  componentWillMount() {
     this.props.UserDetail(this.props.match.params.id)
  }
  handleFormChange = e =>{
       const name = e.target.name
       const value = e.target.value
       this.setState({
        [name] :value
      }) 
   }
   handleSubmit = e =>
        {
            this.props.ChangeDetail(this.props.match.params.id, this.state.email, this.state.name)
        }

  render() {
    return (
        <div className="card">
            <div className="card-header">
                <strong>Cập nhật thông tin nhân viên <i>{this.props.user.username}</i></strong>
            </div>
            <div className="card-body">
                <form className="form-horizontal">
                    <div className="form-group row">
                        <label className="col-md-3 col-form-label" htmlFor="hf-email">Email</label>
                        <div className="col-md-9">
                        <input className="form-control"  type="email" name="email" onChange={this.handleFormChange} autoComplete="email" placeholder={this.props.user.email}/>
                        <span className="help-block">Please enter your email</span>
                        </div>
                    </div>
                    <div className="form-group row">
                        <label className="col-md-3 col-form-label" htmlFor="hf-password">Name</label>
                        <div className="col-md-9">
                            <input className="form-control" type="text" name="name" onChange={this.handleFormChange} placeholder={this.props.user.name}/>
                            <span className="help-block">Please enter your name</span>
                        </div>
                    </div>
                </form>
            </div>
            <div className="card-footer">
                <button className="btn btn-sm btn-primary" onClick={this.handleSubmit}>
                    <i className="fa fa-dot-circle-o"></i> Submit</button>
            </div>
        </div>
    )
  }
}

const mapStateToProps = (state) => ({
    user: state.data.user
  })

function mapDispatchToProps(dispatch) {
    return {
        UserDetail : (id) => {
            dispatch(user_detail(id))
        },
        ChangeDetail : (id, email, name) => {
            dispatch(change_detail(id, email, name))
        }
    }
}
  
export default connect(mapStateToProps, mapDispatchToProps)(UserDetail)