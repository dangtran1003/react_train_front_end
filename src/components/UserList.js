import React, {Component} from 'react'
// import { connect } from "react-redux";
import PropTypes from 'prop-types'
import '../App.css'
import User from './User'
import {user_list, user_add} from '../actions/user'
import { connect } from 'react-redux';

export class UserList extends Component{
    state = {
      username : '',
      email : '',
      name : '',
      error: '', 
      key_search: ''
    }

    handleSubmit = e => {
      e.preventDefault()
      if (this.state.username.length !== 0 && this.state.email.length !== 0)
        this.props.dispatch(user_add(this.state.username, this.state.email, this.state.name))
      else 
        {
          this.setState({error:'Thiếu username hoặc password!'})
          alert('Thiếu username hoặc password!')
        }
    }

    handleFormChange = e =>{
      const name = e.target.name
      const value = e.target.value
      this.setState({
        [name] :value
      })
      if (name === 'key_search')
      {
        this.props.dispatch(user_list(value,this.props.match.params.page-1,10))
      }
    }
    static propTypes = {
        users : PropTypes.array,
        total_users: PropTypes.array
    }

    static defaultProps = {
      users : [],
      total_users: [1]
    }

    componentDidMount() {
        // this.props.dispatch(user_list(null,this.props.match.params.page-1,10))
    }

    // componentWillReceiveProps(nextProps) {
    //   console.log('DEBUG')
    //   console.log(nextProps.users)
    // }

    render(){
        return (
          <div className="card">
          <div className="card-header">
            <i> Danh sách người dùng</i>
            </div>
           
            {this.props.error && alert(this.props.error)}
          <div className="card-body">
          <input className="form-control" type="text" name="key_search" onChange={this.handleFormChange} placeholder="Key search"/>
          <br></br>
            <table className="table table-responsive-sm table-bordered table-striped table-sm">
              <thead>
                <tr>
                  <th>Username</th>
                  <th>Name</th>
                  <th>Email</th>
                  <th>Date registered</th>
                </tr>
              </thead>
              <tbody>
                {
                   this.props.users.map((user, key) => {
                    const {id, username, email, name} = user
                    return (
                      <User username={username} email={email} name = {name} key={key} id={id}></User>
                    )
                  })
                }
                <tr>
                  <td><input className="form-control" name="username" type="text" placeholder="Enter your username" onChange={this.handleFormChange}></input></td>
                  <td><input className="form-control" name="name" type="text" placeholder="Enter your name" onChange={this.handleFormChange}></input></td>
                  <td><input className="form-control" name="email" placeholder="Enter your email" onChange={this.handleFormChange} type="email"></input></td>
                  <td>
                      <button href="#" className="btn btn-success" style={{margin:4}} onClick={this.handleSubmit}><i className="icons font-l d-block mt-0 cui-check" ></i></button>
                      <button href="#" className="btn btn-danger"><i className="icons font-l d-block mt-0 cui-trash"></i></button>
                  </td>
                </tr>
              </tbody>
            </table>
            <nav>
              <ul className="pagination">
                <li className="page-item">
                  <a className="page-link" href={'list/'+(Number(this.props.match.params.page)>1 ? (Number(this.props.match.params.page)-1).toString() : "") }>Prev</a>
                </li>
                <li className={"page-time" +(this.props.match.params.page === 1 ? " active" : "") }>
                  <a className="page-link" href="list/1">1</a>
                </li>
                {this.props.pages.map((page,key)=>{
                  return (
                    <li className={"page-time" +(this.props.match.params.page === page ? " active" : "") } key = {key}>
                  <a className="page-link" href={'list/' + page.toString()}>{page}</a>
                </li>
                  )
                })}
                <li className="page-item">
                  <a className="page-link" href={'list/'+(Number(this.props.match.params.page) +1).toString()}>Next</a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
        )
    }
}

const mapStateToProps = (state) => ({
  users: state.data.users,
  pages: state.data.pages,
  error: state.data.error
})

export default connect(mapStateToProps, null)(UserList)