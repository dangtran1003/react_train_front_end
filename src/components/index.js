import {connect} from 'react-redux'
import UserList from './UserList'

export default connect(({users}) => ({users}))(UserList)