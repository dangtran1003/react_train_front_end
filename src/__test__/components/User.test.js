import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import React from 'react'
import { User } from '../../components/User';


const props = { username : 'trandang',
    name : 'teko vietnam',
    email : 'dang.th@teko.vn',
    id: 1}
describe('Components > User', () => {
    // Snapshot use renderer
   
    it('Render a snapshot for Link use enzyme', () => {
    const tree = shallow(<User {...props}/>);
        console.log(tree.state())
        // expect(tree.find(User)).to.have.lengthOf(1);
        expect(shallowToJson(tree)).toMatchSnapshot();
    })
})