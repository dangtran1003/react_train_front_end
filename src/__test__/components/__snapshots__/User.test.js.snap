// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > User Render a snapshot for Link use enzyme 1`] = `
<tr>
  <td>
    trandang
  </td>
  <td>
    teko vietnam
  </td>
  <td>
    dang.th@teko.vn
  </td>
  <td
    style={
      Object {
        "width": 100,
      }
    }
  >
    <a
      className="btn btn-info"
      href="/modify/1"
      role="button"
      style={
        Object {
          "margin": 4,
        }
      }
    >
      <i
        className="icons font-l d-block mt-0 cui-wrench"
      />
    </a>
    <button
      className="btn btn-danger"
      value={1}
    >
      <i
        className="icons font-l d-block mt-0 cui-trash"
      />
    </button>
  </td>
</tr>
`;
